pybamm/models/full_battery_models/lithium_ion/basic_dfn_half_cell.py

```diff
@@ -0,0 +1,420 @@
+        eps_n = pybamm.PrimaryBroadcast(
+            pybamm.Parameter("Negative electrode porosity"), "negative electrode"
+        )
+        eps_s = pybamm.PrimaryBroadcast(
+            pybamm.Parameter("Separator porosity"), "separator"
+        )
+        eps_p = pybamm.PrimaryBroadcast(
+            pybamm.Parameter("Positive electrode porosity"), "positive electrode"
+        )
+
+        if working_electrode == "negative":
+            eps = pybamm.Concatenation(eps_n, eps_s)
+            tor = pybamm.Concatenation(eps_n ** param.b_e_n, eps_s ** param.b_e_s)
+        else:
+            eps = pybamm.Concatenation(eps_s, eps_p)
+            tor = pybamm.Concatenation(eps_s ** param.b_e_s, eps_p ** param.b_e_p)
+
+        # Interfacial reactions
+        # Surf takes the surface value of a variable, i.e. its boundary value on the
+        # right side. This is also accessible via `boundary_value(x, "right")`, with
+        # "left" providing the boundary value of the left side
+        c_s_surf_n = pybamm.surf(c_s_n)
+        c_s_surf_p = pybamm.surf(c_s_p)
+
+        if working_electrode == "negative":
+            j0_n = param.j0_n(c_e_n, c_s_surf_n, T) / param.C_r_n
+            j_n = (
+                2
+                * j0_n
+                * pybamm.sinh(
+                    param.ne_n / 2 * (phi_s_n - phi_e_n - param.U_n(c_s_surf_n, T))
+                )
+            )
+            j_s = pybamm.PrimaryBroadcast(0, "separator")
+            j_p = pybamm.PrimaryBroadcast(0, "positive electrode")
+            j = pybamm.Concatenation(j_n, j_s)
+        else:
+            j0_p = param.gamma_p * param.j0_p(c_e_p, c_s_surf_p, T) / param.C_r_p
+            j_p = (
+                2
+                * j0_p
+                * pybamm.sinh(
+                    param.ne_p / 2 * (phi_s_p - phi_e_p - param.U_p(c_s_surf_p, T))
+                )
+            )
+            j_s = pybamm.PrimaryBroadcast(0, "separator")
+            j_n = pybamm.PrimaryBroadcast(0, "negative electrode")
+            j = pybamm.Concatenation(j_s, j_p)
+
+        ######################
+        # State of Charge
+        ######################
+        I = param.dimensional_current_with_time
+        # The `rhs` dictionary contains differential equations, with the key being the
+        # variable in the d/dt
+        self.rhs[Q] = I * param.timescale / 3600
+        # Initial conditions must be provided for the ODEs
+        self.initial_conditions[Q] = pybamm.Scalar(0)
+
+        ######################
+        # Particles
+        ######################
+
+        # The div and grad operators will be converted to the appropriate matrix
+        # multiplication at the discretisation stage
+        N_s_n = -param.D_n(c_s_n, T) * pybamm.grad(c_s_n)
+        N_s_p = -param.D_p(c_s_p, T) * pybamm.grad(c_s_p)
+        self.rhs[c_s_n] = -(1 / param.C_n) * pybamm.div(N_s_n)
+        self.rhs[c_s_p] = -(1 / param.C_p) * pybamm.div(N_s_p)
```

Comment:
only need to do one of these depending on what is the working electrode

Reply:
Done